compiler/rustc_type_ir/src/structural_impls.rs

```diff
@@ -165,6 +166,90 @@ impl<I: Interner, T: TypeVisitable<I>, Ix: Idx> TypeVisitable<I> for IndexVec<Ix
     }
 }
 
+///////////////////////////////////////////////////
+//  Debug impls
+
+pub trait InferCtxtLike<I: Interner> {
+    fn universe_of_ty(&self, ty: I::InferTy) -> Option<UniverseIndex>;
+    // FIXME(BoxyUwU): why are lifetimes/consts inconsistent with tys
+    fn universe_of_ct(&self, ct: I::Const) -> Option<UniverseIndex>;
+    fn universe_of_lt(&self, lt: I::Region) -> Option<UniverseIndex>;
+}
+
+impl<I: Interner> InferCtxtLike<I> for core::convert::Infallible {
+    fn universe_of_ty(&self, _ty: <I as Interner>::InferTy) -> Option<UniverseIndex> {
+        unreachable!()
```

Comment:
Minor: 

```suggestion
        match *self {}
```

Looks more robust to me and explicitly shows the reason *why* this is and should be unreachable.